### PR TITLE
Support augmented global properties

### LINKED ISSUE #233: Support augmented global properties
Vue has something like this: https://vuejs.org/guide/typescript/options-api.html#augmenting-global-properties

@antfu suggested to try out to support this, so plugin authors can define their own TypeScript definitions and `eslint-define-config` would then just use these.

So code in the specific plugin projects would look like this:

```ts
import type { DeprecationRuleConfig } from './rules/deprecation/deprecation';

declare module 'eslint-define-config' {
  interface Rules {
    /**
     * Do not use deprecated APIs.
     *
     * @see [deprecation](https://github.com/gund/eslint-plugin-deprecation)
     */
    'deprecation/deprecation'?: DeprecationRuleConfig;
  }
}
```

```ts
import type { AdjacentOverloadSignaturesRuleConfig } from './rules/typescript-eslint/adjacent-overload-signatures';
// ...

declare module 'eslint-define-config' {
  interface Rules {
    /**
     * Require that function overload signatures be consecutive.
     *
     * @see [adjacent-overload-signatures](https://typescript-eslint.io/rules/adjacent-overload-signatures)
     */
    '@typescript-eslint/adjacent-overload-signatures'?: AdjacentOverloadSignaturesRuleConfig;
    // ...
  }
}
```

### PR DESCRIPTION
**Pull request by Shinigami92**
closes #233

### CODE REVIEW

**Comment by Shinigami92** [comment]
I published a [v2.0.0-alpha.0](https://www.npmjs.com/package/eslint-define-config/v/2.0.0-alpha.0) for trying it out

File: README.md [hunk 1]
```diff
@@ -97,6 +101,42 @@ _Click on the thumbnail to play the video_
...
+        {
+          default?: 'array' | 'generic' | 'array-simple';
+          readonly?: 'array' | 'generic' | 'array-simple';
+        },
+      ]
+    >;
+
+    // ... more Rules
+  }
+}
```
[the remark this answers is not in the document]
Comment (Shinigami92) [reply README.md:137]: How can we then map such rule values into `RuleConfig`?
What with rules that do not have config? Are they just `[]`?

[the remark this answers is not in the document]
Comment (antfu) [reply README.md:137]: https://www.typescriptlang.org/play?#code/FAehAIBkEsCNwMIHsAmBTYoIE0kFdwB3aAGxPADs00Vw0APAByQCcAXcAQwroFtG2AT3DQKbNCwBmnAMZpwk1uDwBnCSvBskXPAHNeaMTSx0mrDqPFTZ8hKq28ASnhJoA8gOhIKGgN4BfTCFGeWdXSDQANzRyAF5wAAZwAB9wAEYU8AAmTIByJElJXLzCThYKYtTciRZWXIBuYGD5AAUWNBCKFAAeABU8RldTcS6NbkEAbQBdABpwAEEUdC6APnB4gAoNgH0ALgWlwxQ5gDoz7bT9-sG0AEp1tfH7hhGUDQ2zk72RCkkJcEcaBULjY91ijwowgA-ACgSDwPsqNEWI1mgCXGgItESPMuh42F4fN18YSNC8jmNIdN1lwqVM1vE2h0jt0WmUCZwSMTPN4VCs5mFMVEYitUYIQujXABRMQsQTcgm84YU2mTKY08bTBmSoXYzKCrExXEoEm8hWk0VNcWhDHIX7QXTmpXk0aq6nxTX0mmCmVsOVOnyWq0SwU03zgCb0H7gADWaEEBUQ9iQTgxpp8U32grtkgd3TsKgcgvTKkjXsCYHAVYAelDMCYAKpqFiIVAYYDoGQkMrySwSaRyJOFlPFnk+cC+YBVqu5RRIXL7Ca5WBlXJTYD+IA


> Are they just []?

Yes

[the remark this answers is not in the document]
Comment (Shinigami92) [reply README.md:137]: Wow, nice, so good to learn from you :heart: 
Will rework it in some hours :rocket: 

/ping @azat-io pig to notify you about that change

[the remark this answers is not in the document]
Comment (Shinigami92) [reply README.md]: [v2.0.0-alpha.1](https://www.npmjs.com/package/eslint-define-config/v/2.0.0-alpha.1) published

**Comment by Shinigami92** [comment]
[`f23439b`](https://github.com/Shinigami92/eslint-define-config/pull/238/commits/f23439b6e7e071f6e8419ec3536d6394acef7713) was published as [v2.0.0-alpha.2](https://www.npmjs.com/package/eslint-define-config/v/2.0.0-alpha.2)

**Comment by Shinigami92** [comment]
[v2.0.0-alpha.3](https://www.npmjs.com/package/eslint-define-config/v/2.0.0-alpha.3) has now `CustomExtends` instead of `CustomExtensions`

**Pull request merged**